include/cxx.h

```diff
@@ -826,6 +826,93 @@ template <typename T>
 Box<T>::Box(uninit) noexcept {}
 #endif // CXXBRIDGE1_RUST_BOX
 
+#ifndef CXXBRIDGE1_RUST_OPTION
+template <typename T>
+class Option final {
+public:
+  Option() noexcept;
+  Option(Option&&) noexcept;
+  Option(T&&) noexcept;
+  ~Option() noexcept;
+
+  const T *operator->() const;
+  const T &operator*() const;
+  T *operator->();
+  T &operator*();
+
+  Option<T>& operator=(Option&&) noexcept;
+
+  bool has_value() const noexcept;
+  T& value() noexcept;
+  void reset();
+  void set(T&&) noexcept;
+private:
+  void* empty_;
```

Comment:
In this update, I decided to never access the union directly from C++ and instead always  access it in Rust. I then realizes that means I don't need the union at all on the cpp side? But maybe it's more clear if I keep it regardless